src/if_statements.py

```diff
@@ -118,45 +143,80 @@ def format(self, fmt: Formatter) -> str:
             for (switch, case) in self.context.case_nodes[self.node]:
                 case_str = f"case {case}" if case != -1 else "default"
                 switch_str = f" // switch {switch}" if switch != 0 else ""
-                lines.append(fmt.indent(self.indent, f"{case_str}:{switch_str}"))
+                lines.append(fmt.indent(-1, f"{case_str}:{switch_str}"))
         if self.node in self.context.goto_nodes:
             lines.append(f"{label_for_node(self.context, self.node)}:")
         return "\n".join(lines)
 
 
-Statement = Union[SimpleStatement, IfElseStatement, LabelStatement]
+Statement = Union[
+    SimpleStatement,
+    IfElseStatement,
+    LabelStatement,
+    SwitchStatement,
+]
 
 
 @attr.s
 class Body:
     print_node_comment: bool = attr.ib()
     statements: List[Statement] = attr.ib(factory=list)
 
-    def add_node(self, node: Node, indent: int, comment_empty: bool) -> None:
+    def extend(self, other: "Body") -> None:
+        """Add the contents of `other` into ourselves"""
+        self.print_node_comment |= other.print_node_comment
+        self.statements.extend(other.statements)
+
+    def add_node(self, node: Node, comment_empty: bool) -> None:
         assert isinstance(node.block.block_info, BlockInfo)
         to_write = node.block.block_info.to_write
         any_to_write = any(item.should_write() for item in to_write)
 
         # Add node header comment
         if self.print_node_comment and (any_to_write or comment_empty):
-            self.add_comment(indent, f"Node {node.name()}")
+            self.add_comment(f"Node {node.name()}")
         # Add node contents
         for item in node.block.block_info.to_write:
             if item.should_write():
-                self.statements.append(SimpleStatement(indent, item))
+                self.statements.append(SimpleStatement(item))
 
     def add_statement(self, statement: Statement) -> None:
         self.statements.append(statement)
 
-    def add_comment(self, indent: int, contents: str) -> None:
-        self.add_statement(SimpleStatement(indent, f"// {contents}"))
+    def add_comment(self, contents: str) -> None:
+        self.add_statement(SimpleStatement(f"// {contents}"))
 
     def add_if_else(self, if_else: IfElseStatement) -> None:
+        if if_else.if_body.ends_in_jump():
+            # Transform `if (A) { B; return C; } else { D; }`
+            # into `if (A) { B; return C; } D;`,
+            # which reduces indentation to make the output more readable
+            self.statements.append(attr.evolve(if_else, else_body=None))
+            if if_else.else_body is not None:
+                self.extend(if_else.else_body)
+            return
+
         self.statements.append(if_else)
 
+    def add_switch(self, switch: SwitchStatement) -> None:
+        self.add_statement(switch)
```

Comment:
(in preparation for future changes?)

Reply:
Yeah - this fn & `SwitchStatement` *are* used here, but they could probably be replaced by a plain `Body`.